### PR TITLE
feat(content-utils): Add hooks for content Git lifecycle

### PR DESCRIPTION
**Pull request by Fryuni**
- Simplified and refactored hook provider plugin
- Add support for inter-integration hooks on collections Git lifecycle

### CODE REVIEW

**Review by Adammatthiesen — APPROVED** [review]
Looking pretty good!

**Review by dreyfus92 — APPROVED** [review]
LGTM ✅

File: docs/src/content/docs/content-utils/git-time.mdx [hunk 1]
```diff
@@ -56,3 +56,98 @@ If the entry has not been commited yet, this function returns the current time.
...
+- Override times for entries.
+
+### `@it/content:git:listed`
+
+This hook is called on `astro build` once the collection entries tracked by Git are listed.
+
+```ts
+// your-integration/index.ts
+import { defineIntegration } from 'astro-integration-kit';
+import '@inox-tools/content-utils';
```
Comment (florian-lefebvre) [inline docs/src/content/docs/content-utils/git-time.mdx:74]: Maybe worth highlighting with EC?

**Review by florian-lefebvre — APPROVED** [review]
I've checked quickly and lgtm

**Review by BryceRussell — APPROVED** [review]
LGTM!

**Pull request merged**